### DISCUSSION TITLE
Fixing replica-weighted canary

### DISCUSSION THREAD

**Discussion by Eslam-mohammed-anwar** [argoproj/argo-rollouts#1771] [Q&A] [answered]
## Summary 
Let me try to explain the downtime we have due to that

We are using traefik ingress and we set canary and stable services without setting TrafficRouting as traefik ingress is not supported by ArgoRollout.
The issue here is that during canary steps , ArgoRollout will inject podHash of new pod in the canary service spec.selector and old podHash in the stable service spec.selector , but the traefik ingress is only sending the traffic to stable service so all the traffic load is going to stable service and no load goes to canary at all in our case.
In that case the traffic will increase in the old pods and old pods are terminated and then will end to a downtime until the canary steps are completed
So if you define stable and canary you should define TrafficRouting
Thank you

What happened/what you expected to happen?
- If we set stableService or CanaryService without specifying TrafficRouting is should follow replica-weighted canary
## Diagnostics

What version of Argo Rollouts are you running?
- v1.1.0 

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.
![Rollout1](https://user-images.githubusercontent.com/89852279/149165843-f012c5cd-e84c-4ead-abbb-826f1edf93e1.jpg)

**Comment by perenesenko** [comment, marked as answer]
@Eslam-mohammed-anwar 
You're trying to configure the basic-canary without the traffic routing.

In this case, you can create one more service that will point to all pods from _old replica_ and _new replica_. Name it as an example the **root-service**. This service will be pointing to the all pods and based on a number of pods launched you will get a traffic splitting, managed by **root-service** instead of the Traefik itself.

Upd:
you have to specify the **root-service** in the ingress backend

**Comment by harikrongali** [comment]
https://github.com/argoproj/argo-rollouts/issues/516 is required for what you need to achieve. Not sure if the workaround provided by @perenesenko works for you.
